### ISSUE TITLE
Kotlin 1.9 metadata support

### ISSUE THREAD

**Issue by emartynov** [google/dagger#3838]
Hi people,

I'm trying the Kotlin 1.9 features in project and the project build breaks.
I assume that Kotlin generates a new metadata (maybe just lift version of language there) and Hilt/Dagger doesn't like it:

```
Caused by: java.lang.IllegalStateException: Unsupported metadata version. Check that your Kotlin version is >= 1.0
	at dagger.internal.codegen.kotlin.KotlinMetadata.metadataOf(KotlinMetadata.java:187)
```

**Comment by emartynov** [comment]
Today I learnt that metadata 0.6.0 dependency has it and it is already in main branch https://github.com/google/dagger/blob/180c6b2a958d800834376c87708313d66f928eb4/WORKSPACE#L248

I'm closing this ticket as redundant and waiting for the new release.

**Comment by bcorso** [comment]
Thanks, seems like more people are hitting this so I'll talk with the team and see if we can get a release out this week.

**Comment by bcorso** [comment]
@emartynov this should now be fixed in [Dagger 2.46](https://github.com/google/dagger/releases/tag/dagger-2.46)
